proof/refine/ARM/Finalise_R.thy

```diff
@@ -3985,11 +3999,72 @@ lemma unbind_notification_corres:
                   split: option.splits)
   done
 
-lemma ntfn_at'_cross_rel:
-  "cross_rel (pspace_aligned and pspace_distinct and ntfn_at t) (ntfn_at' t)"
-  unfolding cross_rel_def state_relation_def
+lemma unbind_maybe_notification_corres:
+  "corres dc
+      (valid_objs and ntfn_at ntfnptr) (valid_objs' and ntfn_at' ntfnptr)
```

Comment:
Since you wrote a nice cross lemma for `ntfn_at`, it would be nice to use it here :)

Reply:
it looks like some tests are failing. Can you have a look later/tomorrow? Also, you need to rebase as well.

Reply:
I've removed `ntfn_at'` in the Haskell guard and used the cross lemma, but now `valid_objs/valid_objs'` will have to be changed to `invs/invs'` to use the cross lemma.

Reply:
hmm. I am a bit surprised to hear that. It is unlikely that we need the whole of `invs`. Maybe just adding `pspace_aligned` and `pspace_distinct` should do?